x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -737,6 +738,25 @@ public PlanFactory visitRerankCommand(EsqlBaseParser.RerankCommandContext ctx) {
         return p -> new Rerank(source, p, inferenceId(ctx.inferenceId), queryText, visitFields(ctx.fields()));
     }
 
+    @Override
+    public PlanFactory visitCompletionCommand(EsqlBaseParser.CompletionCommandContext ctx) {
+
+        Source source = source(ctx);
+
+        if (false == EsqlCapabilities.Cap.COMPLETION.isEnabled()) {
+            throw new ParsingException(source, "COMPLETION is in preview and only available in SNAPSHOT build");
+        }
```

Comment:
nit: I don't think this is actually required - because in non-snapshot builds the parser would not allow the use of `COMPLETION` anyway. we would fail with a parsing error before we get to this point. I see that the most "DEV" commands don't have a check like this in `LogicalPlanBuilder`.

Reply:
You are right. Removed these tests.